modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/index.js

```diff
@@ -0,0 +1,122 @@
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {useEffect, useMemo, useState} from 'react';
+import client from '../../../../apolloClient';
+import {Table} from '../../../../common/components';
+import {Liferay} from '../../../../common/services/liferay';
+import {getAccountUserAccountsByExternalReferenceCode} from '../../../../common/services/liferay/graphql/queries';
+import {ROLE_TYPES} from '../../../../common/utils/constants';
+import TeamMembersTableHeader from './components/Header';
+import {
+	NameColumnType,
+	OptionsColumnType,
+	RoleColumnType,
+	StatusColumnType,
+	SupportSeatColumnType,
+} from './utils/constants/columns-definitions';
+import {getColumnsByUserAccess} from './utils/getColumnsByUserAccess';
+
+const TeamMembersTable = ({project}) => {
+	const [userAccounts, setUserAccounts] = useState([]);
+	const [isLoadingUserAccounts, setIsLoadingUserAccounts] = useState(false);
+
+	useEffect(() => {
+		setIsLoadingUserAccounts(true);
+		const getAccountUserAccounts = async () => {
+			const {data} = await client.query({
+				query: getAccountUserAccountsByExternalReferenceCode,
+				variables: {
+					externalReferenceCode: project.accountKey,
+				},
+			});
+
+			if (data) {
+				const accountUserAccounts = data.accountUserAccountsByExternalReferenceCode.items.reduce(
+					(userAccountsAccumulator, userAccount) => {
+						const currentAccountBrief = userAccount.accountBriefs.find(
+							(accountBrief) =>
+								accountBrief.externalReferenceCode ===
+								project.accountKey
+						);
+						if (currentAccountBrief) {
+							userAccountsAccumulator.push({
+								...userAccount,
+								roles: currentAccountBrief.roleBriefs.map(
+									({name}) => name
+								),
+							});
+						}
+
+						return userAccountsAccumulator;
+					},
+					[]
+				);
+				setUserAccounts(accountUserAccounts);
+			}
+
+			setIsLoadingUserAccounts(false);
+		};
+		getAccountUserAccounts();
+	}, [project.accountKey]);
+
+	const hasAdminAccess = useMemo(() => {
+		const currentUser = userAccounts?.find(
+			({id}) => id === Number(Liferay.ThemeDisplay.getUserId())
+		);
+
+		if (currentUser) {
+			const hasAdminRoles = currentUser?.roles?.some(
+				(role) =>
+					role === ROLE_TYPES.admin.key ||
+					role === ROLE_TYPES.requestor.key
+			);
+
+			return hasAdminRoles;
+		}
+	}, [userAccounts]);
+
+	return (
+		<div className="pt-2">
+			<TeamMembersTableHeader
+				hasAdminAccess={hasAdminAccess}
+				project={project}
+				userAccounts={userAccounts}
+			/>
+
+			<Table
+				className="border-0 cp-team-members-table"
+				columns={getColumnsByUserAccess(hasAdminAccess)}
```

Comment:
Should be good pass the return to a variable and use it here
```suggestion
				columns={columnsByUserAccess}
```